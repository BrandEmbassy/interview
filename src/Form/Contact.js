import util from '../util'

export default [
  {
    form: 'contact',
    fields: ["fullName", "bio", "tel", "email"],
    validate: (values) => {
      const errors = {}

      if (!values.fullName) {
        errors.fullName = 'Required'
      }

      if (!util.isPhone(values.tel)) {
        errors.tel = 'Wrong format'
      }

      if (!util.isEmail(values.email)) {
        errors.email = 'Wrong format'
      }

      return errors
    }
  },
  function({ contacts, selectedId }) {
    return {
      initialValues: contacts.find((contact) => contact.id === selectedId)
    }
  }
]
