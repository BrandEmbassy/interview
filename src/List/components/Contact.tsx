import * as React from 'react';
import { Route, Link } from 'react-router-dom';
import * as cx from 'classnames';
import { Contact as ContactType } from '../../state/types';

interface ContactProps {
  contact: ContactType;
}

const Contact: React.SFC<ContactProps> = ({ contact: { id, name } }) => {
  return (
    <Route
      path={`/contact/${id}`}
      children={({ match }) => (
        <Link to={`/contact/${id}`}>
          <div className={cx('item', { 'item--active': !!match })}>
            <div className="in">
              <div className="profile-pic" />
              {name}
            </div>
          </div>
        </Link>
      )}
    />
  );
};

export default Contact;
