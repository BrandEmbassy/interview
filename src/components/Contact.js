import React from "react";

const Contact = ({ name }) => (
  <div className="item">
    <div className="in">
      <div className="profile-pic" />
      {name}
    </div>
  </div>
);

export default Contact;
