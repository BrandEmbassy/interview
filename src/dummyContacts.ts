import { Contact } from './state/types';

const contacts: Contact[] = [
  {
    id: 1,
    name: 'Janko Mrkva',
    bio: 'Possimus velit est voluptatem error.',
    tel: '765789556',
    email: 'fdsf@sdf.sdf',
  },
  {
    id: 2,
    name: 'Patrik Vrbovsky',
    bio:
      'Ipsum qui maxime ad officiis. Qui suscipit accusamus ex porro ullam. Natus velit quis est quos quod dignissimos sunt. Placeat placeat vero. Deleniti rerum distinctio. Sunt quidem eos et et sed suscipit aut voluptate.',
    tel: '678456445',
    email: 'ooo@bar.aaa',
  },
  {
    id: 3,
    name: 'Tomáš Jedno',
    bio:
      'Qui nihil fugit qui inventore ipsum non veritatis. Velit fugiat et. Cupiditate modi itaque nemo et quibusdam odio. Inventore veritatis voluptas repudiandae neque ipsam officia maiores. Itaque fugit corrupti ut perferendis libero praesentium est magnam.',
    tel: '712945023',
    email: 'bla@foo.cz',
  },
];

export default contacts;
