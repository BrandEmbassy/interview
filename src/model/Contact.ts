
interface Contact {
    id: number
    fullName: string
    bio: string
    phone: number
    email: string
}

export = Contact