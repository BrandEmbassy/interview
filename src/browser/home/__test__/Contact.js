import React from 'react';
import test from 'ava';
import { mount } from 'enzyme';
import jsdom from 'jsdom';
import Contact from '../Contact.react';

const doc = jsdom.jsdom('<!doctype html><html><body></body></html>');
global.document = doc;
global.window = doc.defaultView;

test('renders name correctly', t => {
  const name = 'John Doe';
  const el = mount(<Contact name={name} />);

  t.true(el.contains(
    <p className="name">{name}</p>
  ));
});

test('renders editing state correctly', t => {
  const el = mount(<Contact editing />);

  t.true(el.find('input').length === 3);
  t.true(el.find('textarea').length === 1);
});

test('renders disabled button correctly', t => {
  const el = mount(<Contact editing />);

  t.true(el.find('.button.button--positive').is('[disabled]'));
});

test('renders email error correctly', t => {
  const email = 'hi&brandembassy.com';
  const el = mount(<Contact editing email={email} />);

  t.true(el.contains(
    <span className="error-msg">Email is not a valid email</span>
  ));
  t.true(el.find('.email.error').length === 1);
});
