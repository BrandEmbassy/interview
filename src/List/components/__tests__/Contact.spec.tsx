import * as React from 'react';
import Contact from '../Contact';
import render from '../../../test/render';
import dummyContacts from '../../../dummyContacts';

dummyContacts.forEach(contact => {
  it(`should render contact - ${JSON.stringify(contact, null, 2)}`, () => {
    expect(render(<Contact contact={contact} />)).toMatchSnapshot();
  });
});
