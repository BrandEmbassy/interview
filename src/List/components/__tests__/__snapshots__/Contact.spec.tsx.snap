// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render contact - {
  "id": 1,
  "name": "Janko Mrkva",
  "bio": "Possimus velit est voluptatem error.",
  "tel": "765789556",
  "email": "fdsf@sdf.sdf"
} 1`] = `
<a
  href="/contact/1"
  onClick={[Function]}
>
  <div
    className="item"
  >
    <div
      className="in"
    >
      <div
        className="profile-pic"
      />
      Janko Mrkva
    </div>
  </div>
</a>
`;

exports[`should render contact - {
  "id": 2,
  "name": "Patrik Vrbovsky",
  "bio": "Ipsum qui maxime ad officiis. Qui suscipit accusamus ex porro ullam. Natus velit quis est quos quod dignissimos sunt. Placeat placeat vero. Deleniti rerum distinctio. Sunt quidem eos et et sed suscipit aut voluptate.",
  "tel": "678456445",
  "email": "ooo@bar.aaa"
} 1`] = `
<a
  href="/contact/2"
  onClick={[Function]}
>
  <div
    className="item"
  >
    <div
      className="in"
    >
      <div
        className="profile-pic"
      />
      Patrik Vrbovsky
    </div>
  </div>
</a>
`;

exports[`should render contact - {
  "id": 3,
  "name": "Tom치코 Jedno",
  "bio": "Qui nihil fugit qui inventore ipsum non veritatis. Velit fugiat et. Cupiditate modi itaque nemo et quibusdam odio. Inventore veritatis voluptas repudiandae neque ipsam officia maiores. Itaque fugit corrupti ut perferendis libero praesentium est magnam.",
  "tel": "712945023",
  "email": "bla@foo.cz"
} 1`] = `
<a
  href="/contact/3"
  onClick={[Function]}
>
  <div
    className="item"
  >
    <div
      className="in"
    >
      <div
        className="profile-pic"
      />
      Tom치코 Jedno
    </div>
  </div>
</a>
`;
